### PR TITLE
Add attentive layer to Jepa

### PR DESCRIPTION
**Pull request by antoine-tran**
**What does this PR do? Please describe:**

This PR follows up #889 to add the building blocks (models, builders, loader) for the finetuned JEPA encoder, plus testing scripts of the models in different downstream tasks.

**Does your PR introduce any breaking changes? If yes, please list them:**
List of all backwards-incompatible changes.

**Check list:**
- [ ] Was the content of this PR **discussed and approved** via a GitHub issue? (no need for typos or documentation improvements)
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairseq2/blob/main/CONTRIBUTING.md)?
- [ ] Did you make sure that your **PR does only one thing** instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**?
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you **update the [CHANGELOG](https://github.com/facebookresearch/fairseq2/blob/main/CHANGELOG.md)**? (no need for typos, documentation, or minor internal changes)

### CODE REVIEW

**Comment by cbalioglu** [comment]
I am outside for a couple errands, will review before EOD.

File: src/fairseq2/recipes/jepa/models.py [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+            torch.empty(1, num_pools, self.model_dim, device=device, dtype=dtype)
+        )
+
+        if init_fn:
+            init_fn(self.pool_layer)
+
+    def forward(
+        self, seqs: Tensor, padding_mask: PaddingMask | None
+    ) -> tuple[Tensor, PaddingMask | None]:
+        if self.encoder:
```
Comment (cbalioglu) [inline src/fairseq2/recipes/jepa/models.py:268]: Are we sure that this forward implementation is accurate? In the reference implementation [here](https://github.com/facebookresearch/jepa/blob/main/src/models/attentive_pooler.py#L97-L102), I see that encoder stack is applied *after* cross attention which is the reverse of what we have here.

Comment (antoine-tran) [reply src/fairseq2/recipes/jepa/models.py:268]: the internal code seems to deviate from this

**Review by cbalioglu — APPROVED** [review]
LGTM!

**Pull request merged**